exercise/views.py

```diff
@@ -379,9 +379,24 @@ def get_common_objects(self):
                     'name': name,
                     'content': response.text,
                     'html': 'text/html' in response.headers.get('Content-Type'),
+                    'data': response.content,
                 })
         self.note('models')
 
+    def get(self, request, *args, **kwargs):
+        if request.GET.get("download", False):
+            try:
+                bytedata = self.models[0]['data']
+            except OSError:
+                return HttpResponseNotFound()
+
+            response = HttpResponse(bytedata,
+                content_type="application/octet-stream")
+            response["Content-Disposition"] = 'attachment; filename="{}"'\
+                .format(self.models[0]['name'])
+            return response
+
+        return super().get(self)
```

Comment:
You don't pass `self` explicitly to the `super()` parameters in Python 3. Only the normal method parameters are passed. Don't forget to pass the potential `*args` and `**kwargs`.
```suggestion
        return super().get(request, *args, **kwargs)
```

https://docs.python.org/3/library/functions.html#super